atc/db/resource.go

```diff
@@ -340,6 +341,49 @@ func (r *resource) CreateBuild(ctx context.Context, manuallyTriggered bool, plan
 	return build, true, nil
 }
 
+func (r *resource) CreateInMemoryBuild(context context.Context, plan atc.Plan) (Build, error) {
+	tx, err := r.conn.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer Rollback(tx)
+
+	var nextBuildId int
+	err = psql.Select("nextval('builds_id_seq'::regclass)").RunWith(tx).QueryRow().Scan(&nextBuildId)
+	if err != nil {
+		return nil, err
+	}
+
+	err = createBuildEventSeq(tx, nextBuildId)
+	if err != nil {
+		return nil, err
+	}
+
+	build := inMemoryCheckBuild{
+		id:           nextBuildId,
+		checkable:    r,
+		plan:         plan,
+		createTime:   time.Now(),
+		resourceId:   r.id,
+		resourceName: r.name,
+		conn:         r.conn,
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Fprintf(os.Stderr, "EVAN:create in memory check build\n")
+
+	return &build, nil
+}
+
+func (r *resource) CheckApiEndpoint() string {
+	return fmt.Sprintf("/api/v1/teams/%s/pipelines/%s/resources/%s/check?lidar=true", r.teamName, r.pipelineName, r.name)
```

Comment:
I'm guessing this isn't needed given the usage of it is commented out, but curious what it was for

Reply:
I have deleted it, but not push yet.

Originally, I wanted to let Lidar just post a check API to ATC load balancer , so that checks can be distributed to multiple ATCs. But then I realized API authentication is an issue. So I changed to send check builds to `tracker` directly via a channel.